src/tableselection/mouseselectionhandler.js

```diff
@@ -0,0 +1,162 @@
+	 */
+	constructor( tableSelection, editing ) {
+		/**
+		 * Table selection.
+		 *
+		 * @private
+		 * @readonly
+		 * @member {module:table/tableselection~TableSelection}
+		 */
+		this._tableSelection = tableSelection;
+
+		/**
+		 * A flag indicating that the mouse selection is "active". A selection is "active" if it was started and not yet finished.
+		 * A selection can be "active" for instance if user moves a mouse over a table while holding a mouse button down.
+		 *
+		 * @readonly
+		 * @member {Boolean}
+		 */
+		this.isSelecting = false;
+
+		/**
+		 * Editing mapper.
+		 *
+		 * @private
+		 * @readonly
+		 * @member {module:engine/conversion/mapper~Mapper}
+		 */
+		this._mapper = editing.mapper;
+
+		const view = editing.view;
+
+		// Currently the MouseObserver only handles `mouseup` events.
+		view.addObserver( MouseEventsObserver );
+
+		this.listenTo( view.document, 'mousedown', ( event, domEventData ) => this._handleMouseDown( domEventData ) );
+		this.listenTo( view.document, 'mousemove', ( event, domEventData ) => this._handleMouseMove( domEventData ) );
+		this.listenTo( view.document, 'mouseup', ( event, domEventData ) => this._handleMouseUp( domEventData ) );
+	}
+
+	/**
+	 * Handles starting a selection when "mousedown" event has table cell target.
+	 *
+	 * If no table cell in event target it will clear previous selection.
+	 *
+	 * @param {module:engine/view/observer/domeventdata~DomEventData} domEventData
+	 * @private
+	 */
+	_handleMouseDown( domEventData ) {
+		const tableCell = this._getModelTableCellFromDomEvent( domEventData );
+
+		if ( !tableCell ) {
+			this._tableSelection.clearSelection();
+			this._tableSelection.stopSelection();
+
+			return;
+		}
+
+		this.isSelecting = true;
+		this._tableSelection.startSelectingFrom( tableCell );
+	}
+
+	/**
+	 * Handles updating table selection when "mousemove" event has a table cell target.
+	 *
+	 * Does nothing if no table cell in event target or selection is not started.
+	 *
+	 * @param {module:engine/view/observer/domeventdata~DomEventData} domEventData
+	 * @private
+	 */
+	_handleMouseMove( domEventData ) {
+		if ( !isButtonPressed( domEventData ) ) {
+			this._tableSelection.stopSelection();
+
+			return;
+		}
+
+		const tableCell = this._getModelTableCellFromDomEvent( domEventData );
+
+		if ( !tableCell ) {
+			return;
+		}
+
+		this._tableSelection.setSelectingTo( tableCell );
+	}
+
+	/**
+	 * Handles ending (not clearing) table selection on "mouseup" event.
+	 *
+	 * Does nothing if selection is not started.
+	 *
+	 * @param {module:engine/view/observer/domeventdata~DomEventData} domEventData
+	 * @private
+	 */
+	_handleMouseUp( domEventData ) {
+		if ( !this.isSelecting ) {
+			return;
+		}
+
+		const tableCell = this._getModelTableCellFromDomEvent( domEventData );
+
+		// Selection can be stopped if table cell is undefined.
+		this._tableSelection.stopSelection( tableCell );
+	}
+
+	/**
+	 * Finds model table cell for given DOM event.
```

Comment:
```suggestion
	 * Finds a model table cell for a given DOM event.
```

Reply:
Wasn't there a similar helper in the table cell property utils? (I'm not sure)

Reply:
I couldn't find anything that uses `toModelElement()` from a view element.